dreadnode/cli/docker.py

```diff
@@ -0,0 +1,387 @@
+            if "/" in possible_repo or "/" not in value:
+                repo_path, tag = possible_repo, possible_tag
+
+        if not repo_path:
+            raise ValueError("Invalid Docker image format: missing repository name")
+
+        # 3. Determine the registry and the final repository name
+        registry: str | None = None
+        repository = repo_path
+
+        if "/" not in repo_path:
+            # Case 1: An official image like "ubuntu".
+            repository = f"library/{repo_path}"
+        else:
+            # Case 2: A namespaced path. It could be "dreadnode/image"
+            # or "gcr.io/google-containers/busybox".
+            first_part = repo_path.split("/", 1)[0]
+            if "." in first_part or ":" in first_part:
+                # If the first part has a "." or ":", it's a registry hostname.
+                registry = first_part
+                repository = repo_path.split("/", 1)[1]
+
+        # 4. Default to 'latest' tag if no tag or digest is provided
+        if not tag and not digest:
+            tag = "latest"
+
+        # 5. Construct the full, normalized string for the object's value
+        full_image_str = repository
+        if registry:
+            full_image_str = f"{registry}/{repository}"
+
+        if tag:
+            full_image_str += f":{tag}"
+        if digest:
+            full_image_str += f"@{digest}"
+
+        obj = super().__new__(cls, full_image_str)
+        obj.registry = registry
+        obj.repository = repository
+        obj.tag = tag
+        obj.digest = digest
+
+        return obj
+
+    def __repr__(self) -> str:
+        parts = [f"repository='{self.repository}'"]
+        if self.registry:
+            parts.append(f"registry='{self.registry}'")
+        if self.tag:
+            parts.append(f"tag='{self.tag}'")
+        if self.digest:
+            parts.append(f"digest='{self.digest}'")
+        return f"{self.__class__.__name__}({', '.join(parts)})"
+
+    def with_(
+        self,
+        *,
+        repository: str | Unset = UNSET,
+        registry: str | None | Unset = UNSET,
+        tag: str | None | Unset = UNSET,
+        digest: str | None | Unset = UNSET,
+    ) -> "DockerImage":
+        """
+        Create a new DockerImage instance with updated elements.
+        """
+        new_registry = registry if not isinstance(registry, Unset) else self.registry
+        new_repository = repository if not isinstance(repository, Unset) else self.repository
+        new_tag = tag if not isinstance(tag, Unset) else self.tag
+        new_digest = digest if not isinstance(digest, Unset) else self.digest
+
+        new_image = new_repository
+        if new_registry:
+            new_image = f"{new_registry}/{new_repository}"
+
+        if new_tag:
+            new_image += f":{new_tag}"
+        if new_digest:
+            new_image += f"@{new_digest}"
+
+        return DockerImage(new_image)
+
+
+class DockerContainer(BaseModel):
```

Comment:
I did a quick double check and it seems like this limited object works both for `docker ps` and `docker compose ps`, but I do worry about over-extracting fields we don't currently use at the risk of validation errors.